src/proxy.ts

```diff
@@ -1,12 +1,53 @@
 import express from 'express';
-import { createProxyServer } from 'http-proxy';
+import { ClientRequest } from 'http';
+import { createProxyServer, ProxyReqCallback, ProxyResCallback } from 'http-proxy';
+import * as querystring from 'querystring';
 
-const proxy = createProxyServer({ changeOrigin: true });
+import { ProxyParams } from './stubServer';
 
-// Exported for testing purposes only, see [Jest mock inner function](https://stackoverflow.com/q/51269431)
+// [Body Parsing with express](https://github.com/chimurai/http-proxy-middleware/issues/320)
+const handleBodyParsing: ProxyReqCallback = (proxyReq, req) => {
+  // @ts-ignore
+  const { body } = req;
 
-export const send = (
+  const contentType = proxyReq.getHeader('Content-Type');
+  const writeBody = (_body: string) => {
+    proxyReq.setHeader('Content-Length', Buffer.byteLength(_body));
+    proxyReq.write(_body);
+  };
+
+  if (
+    contentType === 'application/json' ||
+    (Array.isArray(contentType) && contentType.includes('application/json'))
+  ) {
+    writeBody(JSON.stringify(body));
+  }
+
+  if (
+    contentType === 'application/x-www-form-urlencoded' ||
+    (Array.isArray(contentType) && contentType.includes('application/x-www-form-urlencoded'))
+  ) {
+    writeBody(querystring.stringify(body));
+  }
+};
+
+const changeOrigin = (proxyReq: ClientRequest, origin: string | undefined) => {
+  if (origin) {
+    proxyReq.setHeader('Origin', origin);
+  }
+};
+
+const removeSecurityFromCookie: ProxyResCallback = proxyRes => {
```

Comment:
Would be nice to add explanations/links:

```TypeScript
/**
 * https://github.com/http-party/node-http-proxy/issues/1165
 */
const removeSecureAttributeFromCookie: ProxyResCallback = proxyRes => {
  /* eslint-disable no-param-reassign */
  proxyRes.headers['set-cookie'] = proxyRes.headers['set-cookie']?.map(cookie =>
    cookie.replace(/; Secure/, '')
  );
  /* eslint-enable no-param-reassign */
}
```

Then the day the issue is resolved at http-proxy level we can remove this code. Without the link to the http-proxy issue it's harder to do that.

Reply:
Explanation for why it's needed + link to the issue => Done